workflow-service/src/test/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowExecutionAspectTest.java

```diff
@@ -136,6 +136,26 @@ public void ExecuteAroundAdviceWithValidDataTest() {
 		verify(this.workFlowService, times(1)).updateWorkFlow(argThat(w -> w.getStatus().toString().equals(COMPLETED)));
 	}
 
+	@Test
+	void ExecuteAroundAdviceWithInProgressWorkFlowTestWithoutWorkFlowDefinition() {
+		// given
+		WorkContext workContext = new WorkContext();
+		when(this.workFlowDefinitionRepository.findFirstByName(any())).thenReturn(null);
+		ProceedingJoinPoint proceedingJoinPoint = mock(ProceedingJoinPoint.class);
+		WorkFlow workFlow = mock(WorkFlow.class);
+		when(proceedingJoinPoint.getTarget()).thenReturn(workFlow);
+		when(workFlow.getName()).thenReturn(TEST_WORK_FLOW);
+
+		// when
+		WorkReport workReport = this.workFlowExecutionAspect.executeAroundAdvice(proceedingJoinPoint, workContext);
+
+		// then
+		assertNotNull(workReport);
+		assertEquals(workReport.getStatus().toString(), FAILED);
+		assertNotNull(workReport.getError());
+		verify(this.workFlowDefinitionRepository, times(1)).findFirstByName(TEST_WORK_FLOW);
```

Comment:
pls add an assertion of the expected exception to be of type ResourceNotFoundException